cli/src/test/java/io/specmesh/cli/CliFunctionalTest.java

```diff
@@ -0,0 +1,127 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.specmesh.cli;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.is;
+
+import io.specmesh.kafka.DockerKafkaEnvironment;
+import io.specmesh.kafka.KafkaEnvironment;
+import java.util.Collection;
+import java.util.Set;
+import java.util.concurrent.ExecutionException;
+import org.apache.kafka.clients.admin.ListTopicsResult;
+import org.apache.kafka.common.acl.AccessControlEntryFilter;
+import org.apache.kafka.common.acl.AclBinding;
+import org.apache.kafka.common.acl.AclBindingFilter;
+import org.apache.kafka.common.resource.PatternType;
+import org.apache.kafka.common.resource.ResourcePatternFilter;
+import org.apache.kafka.common.resource.ResourceType;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+import picocli.CommandLine;
+
+class CliFunctionalTest {
+
+    private static final String OWNER_USER = "simple.schema_demo";
+
+    @RegisterExtension
+    private static final KafkaEnvironment KAFKA_ENV =
+            DockerKafkaEnvironment.builder()
+                    .withSaslAuthentication(
+                            "admin", "admin-secret", OWNER_USER, OWNER_USER + "-secret")
+                    .withKafkaAcls()
+                    .build();
+
+    /**
+     * Provision all resources apart from schemas. note: Dont specify SR credentials to avoid schema
+     * publishing
+     *
+     * @throws Exception - when broken
+     */
+    @Test
+    void shouldProvisionTopicsAndAclResourcesWithoutSchemas() throws Exception {
+
+        final KafkaProvisionCommand kafkaProvisionCommand = new KafkaProvisionCommand();
+
+        // Given:
+        final CommandLine.ParseResult parseResult =
+                new CommandLine(kafkaProvisionCommand)
+                        .parseArgs(
+                                ("--brokerUrl "
+                                                + KAFKA_ENV.kafkaBootstrapServers()
+                                                + " "
+                                                + "--specmeshFile simple_spec_demo-api.yaml "
+                                                + "--username admin "
+                                                + "--secret admin-secret ")
+                                        .split(" "));
+
+        assertThat(parseResult.matchedArgs().size(), is(4));
+
+        // When:
+        kafkaProvisionCommand.run();
```

Comment:
????   Why not just invoke your main method?

Take a look at [SystemTestExecutorFunctionalTest](https://github.com/creek-service/creek-system-test/blob/main/executor/src/test/java/org/creekservice/api/system/test/executor/SystemTestExecutorFunctionalTest.java) in creek - it uses the application built by the `application` gradle plugin to run the tool in a separate jvm, properly testing the thing works end-to-end.

Reply:
its like this because other commands are going to be added and I'm not sure how the main will work together. I can look at it then